lumigator/python/mzai/backend/backend/services/workflows.py

```diff
@@ -0,0 +1,166 @@
+import asyncio
+import datetime
+import uuid
+from collections.abc import Callable
+from uuid import UUID
+
+import loguru
+from fastapi import BackgroundTasks
+from lumigator_schemas.jobs import (
+    JobEvalLiteCreate,
+    JobInferenceCreate,
+    JobStatus,
+)
+from lumigator_schemas.workflows import WorkflowCreate, WorkflowResponse
+
+from backend.services.datasets import DatasetService
+from backend.services.jobs import JobService
+
+
+class WorkflowService:
+    def __init__(
+        self,
+        job_service: JobService,
+        dataset_service: DatasetService,
+    ):
+        self._job_service = job_service
+        self._dataset_service = dataset_service
+
+    NON_TERMINAL_STATUS = [
+        JobStatus.CREATED.value,
+        JobStatus.PENDING.value,
+        JobStatus.RUNNING.value,
+    ]
+    """list: A list of non-terminal job statuses."""
+
+    # TODO: rely on https://github.com/ray-project/ray/blob/7c2a200ef84f17418666dad43017a82f782596a3/python/ray/dashboard/modules/job/common.py#L53
+    TERMINAL_STATUS = [JobStatus.FAILED.value, JobStatus.SUCCEEDED.value]
```

Comment:
would make sense to define these at the pydantic/schema level since we use them throughout jobs and experiments

Reply:
Probably, but it's a copy-paste of existing code in our main. I'm not against changing it but I'd rather do it via a separate PR